fbgemm_gpu/include/fbgemm_gpu/fbgemm_cuda_utils.cuh

```diff
@@ -2211,4 +2223,11 @@ DEVICE_INLINE float float16_min(float_16 val) {
 #undef min
 #undef max
 
+#ifdef __HIP_PLATFORM_HCC__
+__device__ int __any_sync(uint64_t mask, int predicate) {
+  uint64_t predicate_bit_pattern = __ballot(predicate);
+  return (predicate_bit_pattern & mask) > 0;  
+}
+#endif
```

Comment:
Is this what CUDA natively supports (e.g., in some CUDA headers) but HIP currently does not support? If so, I'd appreciate it if you could add a comment.